core/che-core-api-dto/src/main/java/org/eclipse/che/dto/server/DtoFactory.java

```diff
@@ -386,12 +388,37 @@ public Gson getGson() {
     private <T> DtoProvider<T> getDtoProvider(Class<T> dtoInterface) {
         DtoProvider<?> dtoProvider = dtoInterface2Providers.get(dtoInterface);
         if (dtoProvider == null) {
-            throw new IllegalArgumentException("Unknown DTO type " + dtoInterface);
+            if (!dtoInterface.isInterface()) {
+                throw new IllegalArgumentException(format("Only interfaces can be DTO, but %s is not", dtoInterface));
+            }
+
+            if (checkOnDtoAnnotation(dtoInterface)) {
```

Comment:
as for me `check...` method should throw an exception and `is...` or `has...` should return true or false.